### PR TITLE
gromacs: build with legacy api

### PR DESCRIPTION
**Pull request by junghans**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

For https://github.com/espressopp/espressopp/issues/454

### CODE REVIEW

File: Formula/gromacs.rb [hunk 1]
```diff
@@ -54,6 +54,7 @@ def install
     args = %W[
       -DGROMACS_CXX_COMPILER=#{cxx}
       -DGMX_VERSION_STRING_OF_FORK=#{tap.user}
+      -DGMX_INSTALL_LEGACY_API=ON
```
Comment (carlocab) [inline Formula/gromacs.rb:57]: Does the legacy API replace the modern API?

**Comment by SMillerDev** [comment]
If it's a legacy API that is disabled by upstream by default, I'm not sure brew should enable it regardless.

File: Formula/gromacs.rb (hunk 1 above)
Comment (junghans) [reply Formula/gromacs.rb:57]: There is no new API (yet), the main thing the switch does is to install the header files.

**Comment by junghans** [comment]
> If it's a legacy API that is disabled by upstream by default, I'm not sure brew should enable it regardless.

Most other Gromacs packages have this enabled as it only installs a couple more header files and otherwise Gromacs cannot used a library due to the missing headers.

File: Formula/gromacs.rb (hunk 1 above)
Comment (carlocab) [reply Formula/gromacs.rb:57]: I'm confused: why would you install this without any headers? It seems very odd for the default build to install things without the headers.

**Comment by junghans** [comment]
Commit message updated!

**Comment by junghans** [comment]
> > Most other Gromacs packages have this enabled as it only installs a couple more header files and otherwise Gromacs cannot used a library due to the missing headers.
> 
> Can you ask upstream why they disable this by default? It seems like an oversight that a lot of people are compensating for.

I am part of gromacs upstream as well, but the answer is simply that we haven't made any progress on the new API, see 
https://gitlab.com/gromacs/gromacs/-/issues/3288.

@mabraham @eirrgang @pszi1ard any additional comments on this?

**Comment by carlocab** [comment]
> Commit message updated!

Actually, we put the names of formulae first in our commit messages, so it should be something like
```
gromacs: build with legacy C/C++ API
```
Will fix that up for you.

**Review by carlocab — APPROVED** [review]
Fine with this, but it seems like this should just be the upstream default for now if the library is not usable without this option.

**Comment by carlocab** [comment]
Oops, CI didn't run. githubstatus says degraded performance for GHA.

File: Formula/gromacs.rb (hunk 1 above)
Comment (eirrgang) [reply Formula/gromacs.rb:57]: > why would you install this without any headers?

Primarily, the project provides a command line tool.

Several libraries with various degrees of support are also bundled. One of them (libgmxapi) is enabled by default and installs `gmxapi/` headers. The oldest (`libgromacs`) is primarily for internal use to support the command line tool(s). Some of its headers are available, optionally, but it is not well-supported for use by external software.

**Comment by junghans** [comment]
@carlocab thanks!

**Review by p-linnane — APPROVED** [review]

**Comment by carlocab** [comment]
> these headers are deprecated, unsupported, and unstable.

That makes me inclined to revert this change then. We only ship stable and supported software/features in Homebrew/core.

**Comment by p-linnane** [comment]
This should definitely be reverted based off that information.

**Comment by junghans** [comment]
Well one more important note is, there is no replacement for the legacy headers yet and hence the options for library using Gromacs are use the legacy api or drop Gromacs support completely.

**Comment by carlocab** [comment]
Well, as @mabraham says above, the only supported use of the library is in the command line tools, so that's the only thing we should use the library for.

**Comment by SMillerDev** [comment]
> these headers are deprecated, unsupported, and unstable.

That's a pretty clear indication this should be reverted

**Comment by mabraham** [comment]
That said, the headers do little harm and only to the subset of people who opt into using them and perhaps later run into instability. It's your project, but I'd just leave things as they are. :-)

**Comment by junghans** [comment]
@carlocab @p-linnane could one of your help me to create a Formula for the [votca](https://github.com/votca/votca) package?

**Pull request merged**